Assets/BossRoom/Scripts/Client/Game/Action/FXProjectileTargetedActionFX.cs

```diff
@@ -147,6 +147,19 @@ public override void AnticipateAction()
         {
             base.AnticipateAction();
             PlayFireAnim();
+
+            // see if this is going to be a "miss" because the player tried to click through a wall. If so,
+            // we change our data in the same way that the server will (changing our target point to the spot on the wall)
+            if (Data.TargetIds != null && Data.TargetIds.Length > 0)
```

Comment:
The line of sight check should be done whenever there isn't a target as well right? You can fire a mage shot at a random spot on the level with a wall in between and it'll send the projectile to the mouse position.

Reply:
oops, you're right!

ok